clippy_lints/src/casts/cast_possible_truncation.rs

```diff
@@ -43,8 +43,9 @@ fn apply_reductions(cx: &LateContext<'_>, nbits: u64, expr: &Expr<'_>, signed: b
             BinOpKind::Rem | BinOpKind::BitAnd => get_constant_bits(cx, right)
                 .unwrap_or(u64::max_value())
                 .min(apply_reductions(cx, nbits, left, signed)),
-            BinOpKind::Shr => apply_reductions(cx, nbits, left, signed)
-                .saturating_sub(constant_int(cx, right).map_or(0, |s| u64::try_from(s).expect("shift too high"))),
+            BinOpKind::Shr => apply_reductions(cx, nbits, left, signed).saturating_sub(
+                constant_int(cx, right).map_or(0, |s| u64::try_from(s.rem_euclid(u128::from(u64::MAX) + 1)).unwrap()),
```

Comment:
This seems to be the same as `s as u64`

Reply:
yeah, they're the same. The only benefit this way is to pass the dogfood test.

Reply:
You can add an `#[allow]` if it triggers a lint, but it might be preferable to do something like `constant_int().and_then(|s| u64::try_from(s).ok()).unwrap_or(0)`